### ISSUE TITLE
Because the 'da_trainval_net.py' have 'next()' iteration,the T_domain datasets and S_domain datasets must equivalence？？ 

### ISSUE THREAD

**Issue by Zhen-ao** [tiancity-NJU/da-faster-rcnn-PyTorch#2]

**Comment by tjzjp** [comment]
I have the same problem, have you soleved it?

**Comment by tiancity-NJU** [comment]
I think you are right respect to foggy cityscapes dataset. But we can find it 's unneccessary to align source and target images. So I think there are two ways to deal with the problem.
  1. just exclude some images of larger dataset to ensure the size equivalence.

  2. change `iters_per_epoch = s_train_size / args.batch_size` to record large_iters_per_epoch and small_iters_per_epoch.  During train, you just to init the  dataloader at  small_iters_per_epoch with`data_iter = iter(dataloader)` . finally, you need to shuffle the dataloader

**Comment by viniciusarruda** [comment]
Associated to this issue, when I was trying to replicate the Sim10k to Cityscapes results, the `StopIteration` exception was raised. Since the source domain is the Sim10k and the target domain is the Cityscapes, the `s_train_size` is bigger than `t_train_size`, yielding the error. 

So, handling this issue, I changed the line: 
`iters_per_epoch = int(s_train_size / args.batch_size)`

for this one:
`iters_per_epoch = min(int(s_train_size / args.batch_size), int(t_train_size / args.batch_size))`

But I really want to replicate your Sim10k to Cityscapes results faithfully. How should I proceed ? What did you do to get the paper results?
